src/main/java/org/jabref/logic/util/io/CiteKeyBasedFileFinder.java

```diff
@@ -85,18 +82,13 @@ private boolean matches(String filename, String citeKey) {
     private Set<Path> findFilesByExtension(List<Path> directories, List<String> extensions) throws IOException {
         Objects.requireNonNull(extensions, "Extensions must not be null!");
 
-        BiPredicate<Path, BasicFileAttributes> isFileWithCorrectExtension = (path, attributes) ->
-                !Files.isDirectory(path)
-                        && extensions.contains(FileHelper.getFileExtension(path).orElse(""));
+        BiPredicate<Path, BasicFileAttributes> isFileWithCorrectExtension = (path, attributes) -> !Files.isDirectory(path)
+                && extensions.contains(FileHelper.getFileExtension(path).orElse(""));
 
         Set<Path> result = new HashSet<>();
         for (Path directory : directories) {
             if (Files.exists(directory)) {
-                try (Stream<Path> files = Files.find(directory, Integer.MAX_VALUE, isFileWithCorrectExtension)) {
-                    result.addAll(files.collect(Collectors.toSet()));
-                } catch (UncheckedIOException e) {
-                    throw new IOException("Problem in finding files", e);
-                }
+                result.addAll(Files.find(directory, Integer.MAX_VALUE, isFileWithCorrectExtension).collect(Collectors.toSet()));
```

Comment:
Can you please explain this change and why you removed the try-catch as the point of this PR was to actually catch this `UncheckedIOException`.

Reply:
Your code did not compile, you can't throw an exception inside a lambda/stream. That's impossible. 
I did not realize that the unchecked exception can still occur. But we should just log it. I mean, what else can we do about it?`We can't rethrow it.

Reply:
I just pushed a new solution. I think this kind of catch-rethrow should work.